server/src/main/java/org/opentosca/toscana/plugins/cloudfoundry/CloudFoundryLifecycle.java

```diff
@@ -31,30 +40,43 @@ public boolean checkModel() {
 
     @Override
     public void prepare() {
-        //throw new UnsupportedOperationException();
-
         Map<String, String> properties = context.getProperties().getPropertyValues();
 
         if (!properties.isEmpty()) {
-            String username = properties.get("username");
-            String password = properties.get("password");
-            String organization = properties.get("organization");
-            String space = properties.get("space");
-            String apiHost = properties.get("apihost");
-
-            CloudFoundryConnection cloudFoundryConnection = new CloudFoundryConnection(username, password,
-                apiHost, organization, space);
-
-            //TODO: check how to get used provider or figure out whether it is necessary to know it?
-            provider = new CloudFoundryProvider(CloudFoundryProvider.CloudFoundryProviderType.PIVOTAL);
-            provider.setOfferedService(cloudFoundryConnection.getServices());
+            String username = properties.get(CF_PROPERTY_KEY_USERNAME);
+            String password = properties.get(CF_PROPERTY_KEY_PASSWORD);
+            String organization = properties.get(CF_PROPERTY_KEY_ORGANIZATION);
+            String space = properties.get(CF_PROPERTY_KEY_SPACE);
+            String apiHost = properties.get(CF_PROPERTY_KEY_API);
+
+            if (isNotNull(username, password, organization, space, apiHost)) {
+
+                cloudFoundryConnection = new CloudFoundryConnection(username, password,
+                    apiHost, organization, space);
+
+                //TODO: check how to get used provider or figure out whether it is necessary to know it?
+                provider = new CloudFoundryProvider(CloudFoundryProvider.CloudFoundryProviderType.PIVOTAL);
+                provider.setOfferedService(cloudFoundryConnection.getServices());
+            }
+        }
+    }
+
+    private boolean isNotNull(String... elements) {
+        Boolean isOk = true;
+        for (String el : elements) {
+            if (el == null) {
+                isOk = false;
```

Comment:
you dont need to iterate further if one value is null. The result of this method is false at that point. Therefore you can replace this by `return false;`